main.py

```diff
@@ -64,39 +75,78 @@ def main() -> int:
             HIGH_ANGLE,
             LOW_ANGLE,
             ROTATE_SPEED,
-            detection_to_main_queue,
+            detection_to_data_merge_queue,
+            controller,
+        ),
+    )
+
+    data_merge_process = mp.Process(
+        target=data_merge_worker.data_merge_worker,
+        args=(
+            DELAY,
+            flight_interface_to_data_merge_queue,
+            detection_to_data_merge_queue,
+            merged_to_decision_queue,
+            controller,
+        ),
+    )
+
+    decision_process = mp.Process(
+        target=decision_worker.decision_worker,
+        args=(
+            INITIAL_DRONE_STATE,
+            OBJECT_PROXIMITY_LIMIT,
+            MAX_HISTORY,
+            merged_to_decision_queue,
+            command_to_flight_interface_queue,
             controller,
         ),
     )
 
     flight_interface_process.start()
     detection_process.start()
+    data_merge_process.start()
+    decision_process.start()
 
     while True:
         try:
             flight_interface_data: drone_odometry_local.DroneOdometryLocal = (
-                flight_interface_to_main_queue.queue.get_nowait()
+                flight_interface_to_data_merge_queue.queue.get_nowait()
             )
-
             if flight_interface_data is not None:
                 print(flight_interface_data)
 
             detection_data: lidar_detection.LidarDetection = (
-                detection_to_main_queue.queue.get_nowait()
+                detection_to_data_merge_queue.queue.get_nowait()
             )
-
             if detection_data is not None:
                 print(detection_data)
 
+            merged_data: detections_and_odometry.DetectionsAndOdometry = (
+                merged_to_decision_queue.queue.get_nowait()
+            )
+            if merged_data is not None:
+                print(merged_data)
+
+            command_data: decision_command.DecisionCommand = (
+                command_to_flight_interface_queue.queue.get_nowait()
+            )
+            if command_data is not None:
+                print(command_data)
```

Comment:
don't get from the queues in main - the while loop should just be empty (for now we can except keyboard interrupts to shut down the program)